apps/condo/domains/notification/tasks/deliverMessage.js

```diff
@@ -102,6 +104,33 @@ async function deliverMessage (messageId) {
         return MESSAGE_BLACKLISTED_STATUS
     }
 
+
+    // Trying to call hook to define should the message be delivered or not
+    /** @type {AbstractHooksSet} */
+    let hooks
+    /** @constructor */
+    const hooksClass = get(MESSAGE_META, [message.type, 'hooksClass'])
+    if (hooksClass) {
+        hooks = new hooksClass(message)
```

Comment:
what would happen here if `hooksClass ` is not a constructor ?

Reply:
refactored